src/synthesizer/typescript_synthesizer.rs

```diff
@@ -229,13 +305,24 @@ pub fn to_string_ir(resource_value: &ResourceIr) -> Option<String> {
                 None => String::from("{}"),
                 Some(x) => x,
             };
-
             let false_expr = match to_string_ir(false_expr) {
                 None => String::from("{}"),
                 Some(x) => x,
             };
 
-            Option::Some(format!("({})?{}:{}", bool_expr, true_expr, false_expr))
+            Option::Some(format!(
+                "{} ? {} : {}",
+                bool_expr,
+                true_expr,
+                // Convert "{}" to undefined to avoid type mismatch errors. This works for most cases
+                // but not all, e.g., Type 'undefined' is not assignable to type 'IResolvable | PolicyProperty'.
+                // As of now, the user should manually fix when still seeing type mismatch errors.
+                if false_expr == "{}" {
+                    "undefined".to_string()
+                } else {
+                    false_expr
+                }
```

Comment:
iirc: It's more idiomatic to use `match`. Also, I _think_ the only time `{}` comes about is above in the `None` case -- so we can just replace to undefined there.

Reply:
Thanks. That is indeed cleaner.